tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py

```diff
@@ -0,0 +1,338 @@
+"""
+PV Create/Delete & Memory Leak Test: Test the PVC limit with 3 worker nodes
+with & without any IO create and delete the PVCs and check for memory leak
+TO DO: This Test needs to be executed in Scaled setup,
+Adding node scale is yet to be supported.
+"""
+import logging
+import pytest
+import time
+import random
+
+from concurrent.futures import ThreadPoolExecutor
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from tests import helpers
+from ocs_ci.framework.testlib import scale, E2ETest
+from tests.fixtures import (
+    create_ceph_block_pool, create_rbd_secret,
+    create_project, create_cephfs_secret
+)
+
+log = logging.getLogger(__name__)
+
+
+class BasePvcPodCreateDelete(E2ETest):
+    """
+    Base Class to create/delete PVC and POD
+    """
+    all_pvc_obj = []
+    all_pod_obj = []
+    delete_pod_count = 0
+    test_case = ""
+    timeout = 300
+
+    def pre_req(self):
+        """
+        pre_req def to calculate the median value for the memory leak
+        """
+        log.info(f"Pre-req to eveluate memory leak data "
+                 f"before start of test case")
+        self.median_dict = helpers.memory_leak_median()
+        log.info(f"Median dict values for memory leak {self.median_dict}")
+
+    def create_pvc_pod(self, rbd_sc_obj, cephfs_sc_obj, number_of_pvc, size):
```

Comment:
This function is too long.
Please try to split it to several core function to create PVCs and to create Pods that each one gets arguments to fit your needs

Reply:
soon will provide a patch using functions, which will reduce the code in this function.